arangod/Aql/AqlValue.h

```diff
@@ -272,6 +275,79 @@ struct AqlValue final {
     } longNumberMeta;
     static_assert(sizeof(longNumberMeta) == 16,
                   "VPACK_INLINE_INT64 layout is not 16 bytes!");
+
+    // VPACK_SUPERVISED_SLICE
+    struct {
+      velocypack::Slice toSlice() const noexcept {
+        return velocypack::Slice(reinterpret_cast<uint8_t const*>(
+            pointer + sizeof(arangodb::ResourceMonitor*)));
+      }
+      uint64_t getLength() const noexcept {
+        return velocypack::Slice(pointer + sizeof(arangodb::ResourceMonitor*))
+            .byteSize();
+      }
+
+      uint64_t getOrigin() const noexcept {
+        if constexpr (basics::isLittleEndian()) {
+          return (lengthOrigin & 0x000000000000ff00ULL) >> 8;
+        } else {
+          return (lengthOrigin & 0x00ff000000000000ULL) >> 48;
+        }
+      }
+
+      arangodb::ResourceMonitor* getResourceMonitor() const noexcept {
+        // PD points to [ RM* | payload ... ]
+        // So 'pointer' itself is a pointer to a pointer
+        return *reinterpret_cast<arangodb::ResourceMonitor* const*>(pointer);
+      }
+
+      uint8_t* getPayloadPtr() const noexcept {
+        // payload starts at the 9th byte
+        return pointer + sizeof(arangodb::ResourceMonitor*);
+      }
+      uint64_t lengthOrigin;  // First byte: AqlValueType
+                              // Second byte: Memory Origin
+                              // The following 6 bytes: padding
```

Comment:
Name is misleading since it does not contain the length.

Actually why does it not contain the length, same as the managed slice?

Reply:
Yes, but it _is_ confusing - especially because ATM it does _not_ contain the length. But as mentioned in the other comment, I think we _should_ store the length here.